lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -860,6 +864,31 @@ def _create_deployment_parameters(
                 azure_node_runbook.purchase_plan = self._process_marketplace_image_plan(
                     azure_node_runbook.location, azure_node_runbook.marketplace
                 )
+
+            if azure_node_runbook.marketplace:
```

Comment:
Put the whole logic in the method `_get_image_sriov_capability`, including check it's a distro or not.